packages/fuels-types/src/parse_param.rs

```diff
@@ -29,12 +29,20 @@ impl ParamType {
                     // Try to parse tuple (T, T, ..., T)
                     return ParamType::parse_tuple_param(prop, types);
                 }
+                if prop.type_field.contains("generic") {
```

Comment:
Paranoid sanity check: are we sure `type_field` containing `"generic"` is always because of the `generic <NAME>`? Would it be possible that there's a `type_field` that, for whatever reason, contains `generic` and it's not an actual generic within our context?

Reply:
Great catch! I've rewritten this, take a look when you can